rules/DeadCode/PhpDoc/DeadReturnTagValueNodeAnalyzer.php

```diff
@@ -44,6 +45,11 @@ public function isDead(ReturnTagValueNode $returnTagValueNode, ClassMethod $clas
             return false;
         }
 
+        // in case of void, there is no added value in @return tag
+        if ($returnType instanceof Identifier && $returnType->toString() === 'void') {
+            return true;
+        }
+
```

Comment:
I am on mobile, but just looking at it, line 58 can directy return false;

Reply:
This check the PHP return type void, while lines 58 checks for docblock.